src/Ocean/Domains/rectangular_domain.jl

```diff
@@ -0,0 +1,194 @@
+using MPI
+
+using ClimateMachine.Mesh.Grids:
+    DiscontinuousSpectralElementGrid, polynomialorder
+
+using ClimateMachine.Mesh.Topologies: StackedBrickTopology
+
+#####
+##### RectangularDomain
+#####
+
+struct RectangularDomain{FT, G} <: AbstractDomain
+    grid::G
+    Np::Int
+    Ne::NamedTuple{(:x, :y, :z), NTuple{3, Int}}
+    L::NamedTuple{(:x, :y, :z), NTuple{3, FT}}
+    x::NTuple{2, FT}
+    y::NTuple{2, FT}
+    z::NTuple{2, FT}
+end
+
+Base.eltype(::RectangularDomain{FT}) where {FT} = FT
+
+Base.show(io::IO, domain::RectangularDomain{FT, G}) where {FT, G} = print(
+    io,
+    "RectangularDomain{$FT, $(G.name.wrapper)}:",
+    '\n',
+    "    Np = ",
+    domain.Np,
+    ", Ne = ",
+    domain.Ne,
+    '\n',
+    @sprintf(
+        "    L = (x = %.2e, y = %.2e, z = %.2e)",
+        domain.L.x,
+        domain.L.y,
+        domain.L.z
+    ),
+    '\n',
+    @sprintf(
+        "    x = (%.2e, %.2e), y = (%.2e, %.2e), z = (%.2e, %.2e)",
+        domain.x[1],
+        domain.x[2],
+        domain.y[1],
+        domain.y[2],
+        domain.z[1],
+        domain.z[2]
+    )
+)
```

Comment:
Is it worth battling the formatter a bit here?

Reply:
Might be, since I'm not as wily as you I submit to the formatter's dominance much more quickly

Reply:
When it gets really outrageous, you can [skip formatting](https://domluna.github.io/JuliaFormatter.jl/stable/skipping_formatting/) but we really don't want to abuse that.

Reply:
I don't want to spend time on this now so I will battle the formatter in a later PR when I have more tricks up my sleeve.